lemur/plugins/lemur_acme/plugin.py

```diff
@@ -96,20 +99,26 @@ def request_certificate(acme_client, authorizations, csr):
         OpenSSL.crypto.FILETYPE_PEM, cert_response.body
     ).decode('utf-8')
 
-    pem_certificate_chain = "\n".join(
-        OpenSSL.crypto.dump_certificate(OpenSSL.crypto.FILETYPE_PEM, cert.decode("utf-8"))
-        for cert in acme_client.fetch_chain(cert_response)
-    ).decode('utf-8')
+    full_chain = []
+    for cert in acme_client.fetch_chain(cert_response):
+        chain = OpenSSL.crypto.dump_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+        full_chain.append(chain.decode("utf-8"))
+    pem_certificate_chain = "\n".join(full_chain)
 
-    current_app.logger.debug("{0} {1}".format(type(pem_certificate). type(pem_certificate_chain)))
+    current_app.logger.debug("{0} {1}".format(type(pem_certificate), type(pem_certificate_chain)))
     return pem_certificate, pem_certificate_chain
 
 
-def setup_acme_client():
-    email = current_app.config.get('ACME_EMAIL')
-    tel = current_app.config.get('ACME_TEL')
-    directory_url = current_app.config.get('ACME_DIRECTORY_URL')
-    contact = ('mailto:{}'.format(email), 'tel:{}'.format(tel))
+def setup_acme_client(authority):
+    if not authority.options:
+        raise Exception("Invalid authority. Options not set")
+    options = {}
+
+    for option in json.loads(authority.options):
+        options[option.get("name")] = option.get("value")
```

Comment:
Will `option.get("name")` ever return `None`? This would result in: `options[None] = ...`.